src/Collector/CollectorInterface.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Yiisoft\Yii\Debug\Collector;
+
+/**
+ * Debug data collector responsibility is to collect data during application lifecycle.
+ */
+interface CollectorInterface
+{
+    /**
+     * Called once at application startup.
+     * Any initialization could be done here.
+     */
+    public function startup(): void;
+
+    /**
+     * Called once at application shutdown.
+     * Cleanup could be done here.
+     */
+    public function shutdown(): void;
+
+    /**
+     * Collect data.
+     * @return array data collected
+     */
+    public function collect(): array;
+
+    /**
+     * Dispatch data collected for further processing.
+     * @param mixed ...$payload
+     */
+    public function dispatch(...$payload): void;
```

Comment:
It is `$payload` or `$event`? Is there a need to be able to pass multiple arguments?

Reply:
> It is `$payload` or `$event`?

Some collectors will receive `$event`, other won't do that, e.g. `LogCollector`

> Is there a need to be able to pass multiple arguments?

Yes for first stage of developing. Later it may be removed or replaced to more elegant solution